### PR TITLE
feat(typeahead): show options when focused

### LINKED ISSUE #909: Typeahead styles/component
An emerging need in Encore applications is the ability to provide type-ahead tools for a select box or field. We need to create visual/interactive design for this, and then get it added to the Framework in a way that is flexible and appropriate for use in multiple areas across Encore.

@megalord has some work already put in on this in an application, but not at the Framework level. He's currently using https://select2.github.io/, but we need to vet it for use at the Framework level.

cc @megalord @ElementE @parlarjb

### PR DESCRIPTION
**Pull request by megalord**
Fixes #909 
typeahead

### CODE REVIEW

**Comment by glynnis** [comment]
Design: LGTM

File: src/typeahead/typeahead.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                            ngModelCtrl.$setViewValue(ngModelCtrl.$viewValue);
+                        });
+                    });
+
+                    scope.allowEmpty = function (actual, expected) {
+                        if (expected === EMPTY_KEY) {
+                            return true;
+                        }
+                        actual = $filter('lowercase')(actual);
+                        expected = $filter('lowercase')(expected);
```
Comment (parlarjb) [inline src/typeahead/typeahead.js]: Just a small optimization: Do something like `lower = $filter('lowercase')` outside of the `compile` function, and call out to that each time.

**Comment by parlarjb** [comment]
LGTM! Very nice solution to extending angular-bootstrap's `typeahead`

**Comment by CITguy** [comment]
LGTM

**Pull request merged**